graylog2-server/src/main/java/org/graylog/security/shares/EntitySharesService.java

```diff
@@ -86,37 +86,41 @@ public EntitySharesService(DBGrantService grantService,
      * @param sharingSubject the sharing subject
      * @return the response
      */
-    public EntityShareResponse prepareShare(GRN ownedEntity,
+    public EntityShareResponse prepareShare(Optional<GRN> ownedEntity,
```

Comment:
Using `Optional` as a method parameter is an anti-pattern. The type was designed as a return value. IntelliJ also warns about that.

We should also not make the `ownedEntity` nullable. For the `prepareGenericShare` endpoint, we need to get the grantees and capabilities. Instead of having a nullable `ownedEntity` in `#prepareShare`, which makes the code more complicated, we should create a separate method for that in `EntitySharesService`. That method can then only return the information we need and can return without having an existing entity.

Reply:
Was trying to keep FE changes to a minimum. I think we use nullable fields in other places in the FE.

Reply:
How is not using a nullable `ownedEntity` in the backend related to the frontend?

Reply:
Confused. We are talking about params being neither optional nor nullable, right? 
Personally I find an optional param can be more readable than having multiple methods and then factoring out the commonalities. But I'll follow our coding standard.

Reply:
You now pass a nullable parameter to several methods just to return an empty result. If we don't have a nullable parameter in `#prepareShare`, we don't need all the null checks.

From my perspective, the whole method is about preparing share data for an entity. Making the entity nullable changes the behavior massively. So why not use a separate method for it? We also have a separate API endpoint.